apps/examples/testcase/le_tc/kernel/tc_mqueue.c

```diff
@@ -855,6 +855,79 @@ static void tc_mqueue_mq_timedsend_timedreceive_failurechecks(void)
 	mq_unlink("t_mqueues");
 }
 
+
+static void tc_mqueue_mq_getattr(void)
+{
+	mqd_t mqdes;
+	struct mq_attr attr;
+	struct mq_attr mq_stat;
+	int ret_chk;
+
+	/* Fill in attributes for message queue */
+
+	attr.mq_maxmsg  = 20;
+	attr.mq_msgsize = TEST_MSGLEN;
+
+	mqdes = mq_open("mqgetattr", O_CREAT | O_RDWR, 0666, &attr);
+	TC_ASSERT_NEQ("mq_open", mqdes, (mqd_t)ERROR);
+
+	ret_chk = mq_getattr(NULL, &mq_stat);
+	TC_ASSERT_EQ_CLEANUP("mq_getattr", ret_chk, ERROR, goto errout);
+
+	ret_chk = mq_getattr(mqdes, NULL);
+	TC_ASSERT_EQ_CLEANUP("mq_getattr", ret_chk, ERROR, goto errout);
+
+	ret_chk = mq_getattr(mqdes, &mq_stat);
+	TC_ASSERT_EQ_CLEANUP("mq_getattr", ret_chk, OK, goto errout);
+	TC_ASSERT_EQ_CLEANUP("mq_getattr", mq_stat.mq_maxmsg, attr.mq_maxmsg, goto errout);
+	TC_ASSERT_EQ_CLEANUP("mq_getattr", mq_stat.mq_msgsize, attr.mq_msgsize, goto errout);
+
+	TC_SUCCESS_RESULT();
```

Comment:
You should close and unlink mq opened before exit by success.

Reply:
So in that case I have to use return statement and there is duplication of close and unlink calls too.

Reply:
like this:

mq_close(mqdes);
	mq_unlink("mqgetattr");
	TC_SUCCESS_RESULT();
	return;

errout:
	mq_close(mqdes);
	mq_unlink("mqgetattr");

as it was earlier I changed it after yashwant's comment.

Reply:
I revert it @sunghan-chang

Reply:
Oh. I am so sorry. I though `TC_SUCCESS_RESULT` includes return inside. That's why I left this comment. But it isn't so that previous is ok to me.

Reply:
yes I have kept the same.